libgfortran/libgfortran.h

```diff
@@ -67,8 +67,14 @@ extern float __strtof (const char *, char **);
 #define gfc_strtof __strtof
 extern double __strtod (const char *, char **);
 #define gfc_strtod __strtod
+
 extern long double __strtold (const char *, char **);
+#if defined(__arm__) || defined(__aarch64__) || defined(_ARM_) || defined(_ARM64_)
```

Comment:
It shoud be enough to test only one macro, I guess the `__aarch64__` is the preferred one, but please do your own rearch whether this is true.